core/modules/views/views.install

```diff
@@ -149,6 +149,33 @@ function views_update_1004() {
   }
 }
 
+/**
+ * Set all grid style Views to use the deprecrated table if not set already.
+ */
+function views_update_1005() {
+  $active_storage = config_get_config_storage('active');
+  $names = $active_storage->listAll();
+
+  foreach ($names as $name) {
+    if (substr($name, 0, 11) != 'views.view.') {
+      continue;
+    }
+    $config = config($name);
+    $displays = $config->get('display');
+    foreach ($displays as $display => $display_details) {
+      if (!isset($display_details['display_options']['style_plugin']) || $display_details['display_options']['style_plugin'] != 'grid') {
+        continue;
+      }
+      if (!empty($display_details['display_options']['style_options']['deprecated_table'])) {
+        continue;
+      }
+      $config->set('display.' . $display . '.display_options.style_options.deprecated_table', '1');
```

Comment:
I was always confused as to why we sometimes use `'1'`/`'0'` (which gates saved in the config file as text `"1"`/`"0"`), and other times `1`/`0` (which gates saved in the config file as numeric `1`/`0`): https://github.com/backdrop/backdrop-issues/issues/2425

Reply:
This is common in PHP. Unless the variable is cast to be a specific type, PHP will make a guess.